### PR TITLE
Persist offer

### PR DESCRIPTION
**Pull request by andrei-21**

### CODE REVIEW

File: eel/src/data_store.rs [hunk 1]
```diff
@@ -73,20 +74,48 @@ impl DataStore {
                 snapshot_id,
             ),
         )
-        .map_to_invalid_input("Failed to add new incoming payment to payments db")?;
+			.map_to_invalid_input("Failed to add new incoming payment to payments db")?;
```
Comment (gcomte) [inline eel/src/data_store.rs:77]: Formatting error?

File: eel/src/data_store.rs [hunk 2]
```diff
@@ -73,20 +74,48 @@ impl DataStore {
...
+                tx.execute(
+                    "\
+					INSERT INTO offers (payment_id, pocket_id, fiat_currency, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad)\
+					VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8)",
+                    (payment_id, pocket_id, currency_code, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad),
+                )
+                .map_to_invalid_input("Failed to add new incoming payment to payments db")?;
+            }
+            None => (),
+        };
```
Comment (gcomte) [inline eel/src/data_store.rs:117]: How about an `if let`-statement?

```suggestion
        if let Some(OfferKind::Pocket {
            id: pocket_id,
            exchange_rate:
                ExchangeRate {
                    currency_code,
                    rate,
                    updated_at,
                },
            topup_value_minor_units,
            exchange_fee_minor_units,
            exchange_fee_rate_permyriad,
        }) = offer_kind
        {
            let exchanged_at: DateTime<Utc> = updated_at.into();
            tx.execute(
                    "\
					INSERT INTO offers (payment_id, pocket_id, fiat_currency, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad)\
					VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8)",
                    (payment_id, pocket_id, currency_code, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad),
                )
                .map_to_invalid_input("Failed to add new incoming payment to payments db")?;
        };
```

File: eel/src/migrations.rs [hunk 3]
```diff
@@ -127,3 +128,24 @@ fn migration_05_store_payment_failure_reason(connection: &Connection) -> Result<
...
 }
+
+fn migration_06_store_offer_kind(connection: &Connection) -> Result<()> {
+    connection
+        .execute_batch(
+            "\
+            CREATE TABLE offers (
+                id INTEGER NOT NULL PRIMARY KEY,
+                payment_id INTEGER NOT NULL,
+
```
Comment (gcomte) [inline eel/src/migrations.rs:139]: formatting error?

File: examples/3l-node/cli.rs [hunk 4]
```diff
@@ -625,12 +625,39 @@ fn list_payments(node: &LightningNode) -> Result<(), String> {
...
-        println!();
+        println!("      Offer:        {}", offer_to_string(payment.offer));
     }
 
     Ok(())
 }
 
+fn offer_to_string(offer: Option<OfferKind>) -> String {
+    match offer {
+        Some(OfferKind::Pocket {
```
Comment (gcomte) [inline examples/3l-node/cli.rs:636]: Probably the right way would be to implement the `Display` trait for `OfferKind` and handle the Option here separately

File: eel/src/data_store.rs (hunk 2 above)
Comment (gcomte) [reply eel/src/data_store.rs:117]: ok

File: examples/3l-node/cli.rs (hunk 4 above)
Comment (gcomte) [reply examples/3l-node/cli.rs:636]: Well, `Display` could be implemented within `3L` where `OfferKind` itself is implemented, couldn't it?

File: eel/src/data_store.rs [hunk 5]
```diff
@@ -73,20 +74,48 @@ impl DataStore {
...
+                exchange_fee_rate_permyriad,
+            }) => {
+                let exchanged_at: DateTime<Utc> = updated_at.into();
+                tx.execute(
+                    "\
+					INSERT INTO offers (payment_id, pocket_id, fiat_currency, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad)\
+					VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8)",
+                    (payment_id, pocket_id, currency_code, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad),
+                )
+                .map_to_invalid_input("Failed to add new incoming payment to payments db")?;
```
Comment (dleutenegger) [inline eel/src/data_store.rs:114]: Error message doesn't seem to fit
```suggestion
                .map_to_invalid_input("Failed to add new incoming pocket offer to offers db")?;
```

**Pull request merged**